SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+# SPOOL Improvement Proposal 001
+
+
+## Abstract
+
+This proposal discusses details for allowing the implementation
+of unlimited editions in the SPOOL ownership API.
+
+## Requirements
+
+    - An API user must be able to bulk create a set of unlimited editions for a piece in a single
+      HTTP request
+    - An API user must be able to repeatedly create a set of unlimited editions
+      for a piece using HTTP requests
+    - An API user is not required to know about concurrent edition creation
+      processes running on a piece they try to create editions on
+      (meaning an API user can submit a request for creating editions, and even
+      though this process is still running, can send another request for
+      creating more editions)
+    - An API user must be able to create a distinct set of editions which
+      numbers do neither need to be in order, nor do they need to be consecutive
+      (All these examples shall be valid: [1, 2, 3], [3, 1, 2], [3, 1])
+    - An API user must be able to `freeze` or limit the unlimited editions of a
+      piece using a single HTTP request
+    - An API user must be able to track the creation of editions of a piece in a
+      modern-web kind of fashion (e.g. using websockets, instead of polling)
+
+
+## Problem
+
+Currently when creating editions, a user can make use of two endpoints:
+
+    1. Set `num_editions` when registering a piece `POST /api/pieces`
+    2. Set `num_editions` and `piece_id` using `POST /api/editions`
+
+When calling either of those endpoints, the user currently creates a set of
+limited editions in the range of 0 and the value of `num_editions` and a SPOOL
+ownership action (in form of a Bitcoin transaction) is triggered.
+
+For allowing unlimited editions to be implemented the following changes (that are
+not up for discussion in this SIP) will be made to the SPOOL protocol:
+
+    - When registering editions, a SPOOL `OwnershipEditions` Bitcoin transaction
+      will not be sent to the Bitcoin network immediately
+    - When doing actions(transfer, consign, loan) on a previously
+      registered/created edition, ownership actions on the Bitcoin network will
+      be recorded as usual
+    - When the user decides to limit his unlimited editions (we call this a
+      `freeze` action) a SPOOL `OwnershipEditions` Bitcoin transaction will be
+      sent to the Bitcoin network
+
+As this change needs to be implemented in version 0 of the API, its changes must
+not break the current endpoint's behavior. This exposes the following constraint:
+
+    - Using `num_editions` in `POST /api/pieces` and `POST /api/editions` still
+      needs to register a limited set of editions in a range of 0 to the value
+      of `num _editions`
+
+
+## Solution
+
+### Creation of unlimited editions
+
+Using `POST /api/pieces/1234/editions/` an API user can create a distinct set of unlimited
```

Comment:
Thanks for this question, as I think it exposes the most interesting properties of this change.

First of, lets distinguish here in philosophical and technical reasoning.
I'll only go into details for the technical reasoning here:

The paradigm we adopted for the SPOOL REST API is that it should just be a reflection of what is possible with the SPOOL protocol.
When you register limited editions as of now, this is what happens on the Bitcoin blockchain:

1. Register the piece `OWNERSHIPPIECE`
2. Register the editions under the piece `OWNERSHIPEDITONS`
-- (because of "lazy editions") Wait for the user to transfer an edition --
3. Register the individual edition `OWNERSHIPREGISTER`
4. Register for the edition a transfer `OWNERSHIPTRANSFER`

If we play this through with a reallife example:

1. Tim registers Piece and registers 10 editions `OWNERSHIPPIECE` and `OWNERSHIPEDITONS`
2. Tim decides to transfer edition 5 `OWNERSHIPREGISTER5` and `OWNERSHIPTRANSFER5`

So as you can see, even currently, registering limited editions non-sequentially is a valid technical usecase.
As we want the SPOOL REST API just to be a pretty interface for writing stuff on some blockchain, allowing for the creation of non-sequential editions is just logical.